### PR TITLE
Nlp string implementation

### PR DESCRIPTION
**Pull request by fabregas**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,5 +15,5 @@ before_install:
   - go get github.com/stretchr/testify/assert
 
 script:
-  - go test -v -covermode=count -coverprofile=coverage.out ./nlp_string
-  - goveralls -coverprofile=coverage.out -service travis-ci -repotoken $COVERALLS_TOKEN
+  - go test -v -covermode=count -coverprofile=coverage.out ./core
```
Comment (korobool) [inline .travis.yml]: We have to discuss how to organize this. I think we need come up with some recursive strategy

File: core/strings.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
+//
+// Package strings implements unicode string abstraction
```
Comment (korobool) [inline core/strings.go:2]: I'm wondering is this some kind of agreement how to provide examples in golang?
If not, we can put markdown in the same folder

File: core/strings.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+	s.ridx += 1
+	// padding to 4 bytes per rune
+	// expecting offset returned from regexp.FindReaderIndex equal to rune_offset*4
+	size = 4
+	return
+}
+
+// This internal function convert bytes location to runes location in string
+func (s *String) normalizeRegexpLoc(runes_offset, from, to int) (loc []int) {
+	from_rune := runes_offset + (from / 4)
```
Comment (korobool) [inline core/strings.go:89]: Does this mean that rune is ALWAYS 4 bytes now?

File: examples/tagger_train.go [hunk 4]
```diff
@@ -1,3 +1,5 @@
+// +build ignore
```
Comment (korobool) [inline examples/tagger_train.go:1]: This and a set of other files seem to be not related to String.  This is not the problem, but let's try to keep things atomic.

**Comment by korobool** [comment]
What restrictions and compatibility level does String have with stdlib regex?

File: core/strings.go (hunk 3 above)
Comment (fabregas) [inline core/strings.go:89]: 1) `rune` is an alias to `int32`.
2) `ReadRune` method always returns size=4.. so we expecting offset returned from regexp.FindReaderIndex equal to rune_offset*4

File: core/strings.go (hunk 2 above)
Comment (fabregas) [inline core/strings.go:2]: I think, every package should provide some usage examples.
I didn't see this usage examples for other packages, so wrote it in package docstring.

**Comment by fabregas** [comment]
some methods of String expected compiled `regexp` object as an argument:
`String.Match(re)`
`String.FindFirst(re)`
`String.FindAll(re)`
`String.Replace(re, new)`

.... The most methods in standard regexp package works with `[]byte` and `string`, so we can't use it with String object

**Comment by korobool** [comment]
Looks good for me. Merge needed.
